fastai/distributed.py

```diff
@@ -77,16 +77,16 @@ def _round_to_multiple(number,multiple): return int(math.ceil(number/multiple)*m
 # %% ../nbs/20a_distributed.ipynb 18
 class DistributedDL(TfmdDL):
     "A `TfmdDL` which splits a batch into equal size pieces for each worker"
-    def __init__(self,dl,rank=None,world_size=None):
+    def __init__(self,dl,rank=None,world_size=None,device=None):
         if rank is None: rank=rank_distrib()
         if world_size is None: world_size=num_distrib()
         store_attr()
         if type(dl) == torch.utils.data.DataLoader:
             shuffle = True if eq(type(dl.sampler), torch.utils.data.RandomSampler) else False
             self.dl = DataLoader(dataset=dl.dataset, bs=dl.batch_size, num_workers=dl.num_workers, \
                 pin_memory=dl.pin_memory, timeout=dl.timeout, shuffle=shuffle, drop_last=dl.drop_last, persistent_workers=dl.persistent_workers)
-        self.bs,self.device,self.drop_last,self.dataset,fake,self.num_workers,self.offs,self.pin_memory = \
-            attrgetter('bs','device','drop_last','dataset','fake_l','num_workers','offs','pin_memory')(self.dl)
+        self.bs,self.drop_last,self.dataset,fake,self.num_workers,self.offs,self.pin_memory = \
+            attrgetter('bs','drop_last','dataset','fake_l','num_workers','offs','pin_memory')(self.dl)
```

Comment:
I just realized that to make this a backwards-compatible and non-breaking change, we should continue to get `self.device` from `self.dl` in the case where device isn't passed in.

Reply:
Fixed with the conditional check below.